### PR TITLE
Set MooseApp executioner right after executioner is created

### LINKED ISSUE #5497: MooseApp's executioner is set at the end of all actions
While ActionWarehouse's executioner is set right after executioner is added. I do not see why this should be. Also when users want to access data in executioner in a kernel constructor, they have to go action warehouse to grab the executioner.

I think with the fix, `EigenKernel::initialSetup()` can be moved into its constructor.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close  #5497.

### CODE REVIEW

**Comment by YaqiWang** [comment]
A better fix is pushed.

File: framework/src/actions/CreateExecutionerAction.C [hunk 1]
```diff
@@ -110,6 +110,7 @@ CreateExecutionerAction::act()
   Moose::setup_perf_log.pop("Create Executioner","Setup");
 
   _awh.executioner() = executioner;
```
Comment (permcody) [inline framework/src/actions/CreateExecutionerAction.C]: Can this be cleaned up then? If we aren't using it we should try deleting it.

File: framework/src/kernels/EigenKernel.C [hunk 2]
```diff
@@ -36,11 +36,6 @@ EigenKernel::EigenKernel(const InputParameters & parameters) :
     _eigen_sys(dynamic_cast<EigenSystem *>(&_fe_problem.getNonlinearSystem())),
     _eigenvalue(NULL)
 {
-}
-
-void
-EigenKernel::initialSetup()
```
Comment (permcody) [inline framework/src/kernels/EigenKernel.C]: Moving all of this to the constructor is OK but means that we are definitely requiring that the Executioner exists before this object is even built!

Comment (YaqiWang) [inline framework/src/kernels/EigenKernel.C]: By setting executioner pointer in `MooseApp`, this is made possible.

File: framework/src/actions/CreateExecutionerAction.C (hunk 1 above)
Comment (YaqiWang) [inline framework/src/actions/CreateExecutionerAction.C]: Not sure, let me remove the accessor in AWH, see how much the damage is.

Comment (permcody) [inline framework/src/actions/CreateExecutionerAction.C]: That would be great, there may be a few `Actions` using it but they could just get it from MooseApp instead.

Comment (YaqiWang) [inline framework/src/actions/CreateExecutionerAction.C]: Actually `Action` constructs a reference of the shared pointer from AWH. If I get rid of AWH accessor, I will need to change MooseApp interface, which will result into a painful set of changes, possibly patching applications. So do we really want to remove the executioner accesso in AWH?

File: framework/src/kernels/EigenKernel.C (hunk 2 above)
Comment (YaqiWang) [inline framework/src/kernels/EigenKernel.C]: Kernel is always built way late. So this is fine.

File: framework/src/actions/CreateExecutionerAction.C (hunk 1 above)
Comment (permcody) [inline framework/src/actions/CreateExecutionerAction.C]: That's fine, I much prefer making two independent changes then one that requires application patches immediately. Just leave the ticket open and we'll adjust the AWH later to remove the other reference.

**Comment by YaqiWang** [comment]
@permcody I kept two versions of getting executioner in MooseApp, so that no application patch is needed. We might remvoe `getExecutioner()` in the future.

**Comment by YaqiWang** [comment]
Stepped on my own foot... I am fixing yak, so that this PR will pass the app tests.

**Comment by YaqiWang** [comment]
More fix in yak needs to be done... This small change is causing me headache.

**Comment by jwpeterson** [comment]
This branch has been moved into upstream as `moose_app_exe_5497` for Yak integration.  @YaqiWang, let me know when Yak master has been updated and I will merge this one.  Thanks.

**Comment by YaqiWang** [comment]
This PR seems not change anything except the interface. But it causes a test in Rattlesnake exodiff. I need more time to investigate why it is happening.

**Comment by YaqiWang** [comment]
Found the problem... @aeslaughter created an new task called `add_ls_kernel` in Yak, which makes eigen kernels being added before postprocessor for eigenvalue is added. It is ok before this PR, because the postprocessor is obtained at eigen kernel initial setup. I guess @aeslaughter did that for debugging purpose.

**Comment by YaqiWang** [comment]
@jwpeterson I need you move my branch into integration once again. I might missed few changes in my previous PR.

**Comment by aeslaughter** [comment]
see #5532

**Pull request closed without merging**